image/image.go

```diff
@@ -213,19 +213,24 @@ func makeFetcher(sto Store, dlOpts *DownloadOptions, db *asserts.Database) *addi
 func installCloudConfig(gadgetDir string) error {
 	var err error
 
-	cloudDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud")
+	cloudDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud/")
 	if err := os.MkdirAll(cloudDir, 0755); err != nil {
 		return err
 	}
 
 	cloudConfig := filepath.Join(gadgetDir, "cloud.conf")
 	if osutil.FileExists(cloudConfig) {
-		dst := filepath.Join(cloudDir, "cloud.cfg")
+		cfgDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud/cloud.cfg.d")
+		if err := os.MkdirAll(cfgDir, 0755); err != nil {
+			return err
+		}
+		dst := filepath.Join(cfgDir, "10_gadget_snap_cloud.cfg")
```

Comment:
@niemeyer it's certainly possible;  I think it's not a likely scenario if the core snap contains the default cloud-init configuration as found in the Ubuntu Cloud image;  this nudges folks into only changing cloud config in the gadget snap (where I think we want it).